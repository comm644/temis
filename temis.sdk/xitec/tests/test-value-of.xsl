<?xml version="1.0" encoding="utf-8"?>
<!-- 

   Copyright (c) 2008 Alexey V. Vasilyev

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.

-->
<xsl:stylesheet
  xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
  xmlns:xitec="xitec.dtd"
                version="1.0">

  <xsl:output method="html"/>

  <xsl:template match="/">

    <xitec:print-mode/>

    doc 
    
    1 will be copied
    <xsl:value-of select="node"/>
    
    2 will be generated with path
    <xsl:value-of select="/object/node"/>

    3 will be generated clean
    <xsl:value-of select="/object/$node"/>

    4 will be copied clean
    <xsl:value-of select="/widget/node"/>
    <xsl:value-of select="/widget/$node"/>

    <xitec:switch-mode mode="widget">
      5 will be copied
      <xsl:value-of select="node"/>
      
      6 will be generated with path
      <xsl:value-of select="/object/node"/>

      7 will be generated clean
      <xsl:value-of select="/object/$node"/>

      8 will be copied clean
      <xsl:value-of select="/widget/node"/>
      <xsl:value-of select="/widget/$node"/>
    </xitec:switch-mode>

    <xitec:switch-mode mode="object">
      9 will be generated
      <xsl:value-of select="node"/>
      
      10 will be generated with path
      <xsl:value-of select="/object/node"/>

      11 will be generated clean
      <xsl:value-of select="/object/$node"/>

      12 will be copied clean
      <xsl:value-of select="/widget/node"/>
      <xsl:value-of select="/widget/$node"/>
    </xitec:switch-mode>
    
  </xsl:template>
</xsl:stylesheet>
