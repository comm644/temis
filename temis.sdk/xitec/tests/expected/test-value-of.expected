<?xml version="1.0" encoding="utf-8" standalone="no"?>
<xsl:stylesheet xmlns:xitec="xitec.dtd" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">

  <xsl:output method="html"/>

  <xsl:template match="/"><xsl:param name="address"/><xsl:param name="realaddress"/><xsl:param name="node"/>

    [xitec:mode=doc]

    doc 
    
    1 will be copied
    <xsl:value-of select="node"/>
    
    2 will be generated with path
    <xsl:call-template name="gen:newline"/><xsl:element name="xsl:value-of"><xsl:attribute name="select"><xsl:apply-templates select="." mode="gen:object-uri"><xsl:with-param name="address" select="$address"/><xsl:with-param name="realaddress" select="$realaddress"/><xsl:with-param name="node" select="$node"/></xsl:apply-templates><xsl:text>/</xsl:text>node</xsl:attribute></xsl:element>

    3 will be generated clean
    <xsl:call-template name="gen:newline"/><xsl:element name="xsl:value-of"><xsl:attribute name="select">$node</xsl:attribute></xsl:element>

    4 will be copied clean
    <xsl:value-of select="node"/>
    <xsl:value-of select="$node"/>

    
      5 will be copied
      <xsl:value-of select="node"/>
      
      6 will be generated with path
      <xsl:element name="xsl:value-of"><xsl:attribute name="select"><xsl:apply-templates select="." mode="gen:object-uri"><xsl:with-param name="address" select="$address"/><xsl:with-param name="realaddress" select="$realaddress"/><xsl:with-param name="node" select="$node"/></xsl:apply-templates><xsl:text>/</xsl:text>node</xsl:attribute></xsl:element>

      7 will be generated clean
      <xsl:element name="xsl:value-of"><xsl:attribute name="select">$node</xsl:attribute></xsl:element>

      8 will be copied clean
      <xsl:value-of select="node"/><xsl:value-of select="$node"/>

    <xsl:text>
      9 will be generated
      </xsl:text><xsl:element name="xsl:value-of"><xsl:attribute name="select">/node</xsl:attribute></xsl:element><xsl:text>
      
      10 will be generated with path
      </xsl:text><xsl:element name="xsl:value-of"><xsl:attribute name="select"><xsl:apply-templates select="." mode="gen:object-uri"><xsl:with-param name="address" select="$address"/><xsl:with-param name="realaddress" select="$realaddress"/><xsl:with-param name="node" select="$node"/></xsl:apply-templates><xsl:text>/</xsl:text>node</xsl:attribute></xsl:element><xsl:text>

      11 will be generated clean
      </xsl:text><xsl:element name="xsl:value-of"><xsl:attribute name="select">$node</xsl:attribute></xsl:element><xsl:text>

      12 will be copied clean
      </xsl:text><xsl:value-of select="node"/><xsl:value-of select="$node"/>
    
  </xsl:template>
</xsl:stylesheet>
